core/state/dump.go

```diff
@@ -34,16 +37,76 @@ type DumpAccount struct {
 	Storage  map[string]string `json:"storage"`
 }
 
+// For output in a collected format, as one large map
 type Dump struct {
 	Root     string                 `json:"root"`
 	Accounts map[string]DumpAccount `json:"accounts"`
 }
 
-func (self *StateDB) RawDump() Dump {
-	dump := Dump{
-		Root:     fmt.Sprintf("%x", self.trie.Hash()),
+// DumpAccountFull is the same as DumpAccount but also with address, for standalone printing
+type DumpAccountFull struct {
+	Address  string            `json:"address"`
+	Balance  string            `json:"balance"`
+	Nonce    uint64            `json:"nonce"`
+	Root     string            `json:"root"`
+	CodeHash string            `json:"codeHash"`
+	Code     string            `json:"code"`
+	Storage  map[string]string `json:"storage"`
+}
```

Comment:
Instead of duplicating everything, you could achieve a similar effect by adding

```
Address *string `json:"address,omitempty"`
```

to `DumpAccount`.